### PR TITLE
Fixed #8 path to procfs is now confugurable

### LINKED ISSUE #8: Path to procfs is not configurable
The path to procfs is currently [hardcoded](https://github.com/intelsdi-x/snap-plugin-collector-load/blob/master/load/load.go#L50) to /proc. This makes it hard (if even possible) to use the plugin when snapd is run in a Docker container.

### PR DESCRIPTION
**Pull request by marcin-krolik**
- "procfs_path" added as configuration item, with default value "/procfs/loadavg"
- namespace is now build staticaly, without reading the file
- updated unit tests

### CODE REVIEW

**Comment by obourdon** [comment]
Hello Marcin,

Just to let you know, I had taken a different approach if you look at

https://github.com/obourdon/snap-plugin-collector-load/tree/fix_4_docker
which depends on
https://github.com/obourdon/snap/tree/fix_4_docker

Using global environment variable for the snapd process itself

Do you think we should keep both method and I need to submit pull requests at some point in time or just stick with your solution ? I do not mind dropping my change at all

Thanks & best regards

**Comment by marcin-krolik** [comment]
Hi Oliver, 
From my perspective it is more clear and clean to provide plugin specific configuration via task manifest dedicated section. I'm not sure if snap framework is the right place for taking care of particular environment variables for some plugins. @jcooklin what do you think?

@obourdon

**Comment by obourdon** [comment]
@marcin-krolik I completely understand and partially agree ;-). Having this also in the main configuration ensures/enforce consistency across all plugins loaded within the same snapd instance for instance. Thanks for all (other) inputs on this.

**Comment by marcin-krolik** [comment]
@obourdon This is good point. Do you think that assuring consistency by tribe functionality would solve this problem? 
I just don't like the idea of "env vars framework driven plugin configuration" :). I would prefer to rely on already implemented configuration mechanisms like global config and task manifest config. Both methods have limitations and there is a trade off between consistency of configuration and ability to load plugins. I thought that removal of global configuration dependency in plugins is more important then eventual consistency across snapd instance. 
Handling configuration of env vars in framework would basically provide same mechanism like global config, but instead of json file direclty provided on snapd startup, one would need to provide env vars setup.

**Comment by obourdon** [comment]
@marcin-krolik Just to make sure I fully understand what you suggest here. Does this mean that you'd rather have some entry like:

`        flProcFS = cli.StringFlag{
                Name:   "procfs-location",
                Usage:  "Location where procfs files can be found (for Docker support). Default: empty string",
                EnvVar: "HOST_PROC",
        }
`

added into snapd.go and in addition to your change ?

This way, global configuration, environment variable and task manifest configuration would be supported.

Of course, it would be assumed that the more specific configuration (task) is taking precedence over global configuration ...

Does this makes sense ?

**Comment by marcin-krolik** [comment]
@obourdon Not at all. I would stay with managing plugin configuration in currently available ways which is global config and task manifest with task manifest as preferred solution. The reason behind chosing task manifest configuration over global config is described in [snap#936](https://github.com/intelsdi-x/snap/issues/936). 
Implementation provided in this PR makes plugin runnable in Docker container (configuration for procfs may be provided in task manifest, if not plugin uses default one) and plugin builds metric catalog statically, so it will load always.

File: README.md [hunk 1]
```diff
@@ -96,9 +105,8 @@ Create a task manifest file (exemplary file in [examples/task/] (https://github.
                 "/intel/procfs/load/scheduling": {}
             },
             "config": {
-                "/intel/mock": {
-                    "password": "secret",
-                    "user": "root"
+                "/intel/procfs/load/": {
+                    "procfs_path": "/var/procfs/loadavg"
```
Comment (elemoine) [inline README.md]: For the CPU (https://github.com/intelsdi-x/snap-plugin-collector-cpu/pull/16) and Swap (https://github.com/intelsdi-x/snap-plugin-collector-swap/pull/8) plugins the option name is `proc_path`. I think the naming should be consistent.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,11 @@ This builds the plugin in `/build/rootfs/`
 `export SNAP_PATH=$GOPATH/src/github.com/intelsdi-x/snap/build`
 * Load the plugin and create a task, see example in [Examples](https://github.com/intelsdi-x/snap-plugin-collector-load/blob/master/README.md#examples).
 
+Plugin may be additionally configured with path to `/proc/loadavg` file. It may be useful for running plugin in Docker container.
+Example configuration provided in [Examples](https://github.com/intelsdi-x/snap-plugin-collector-load/blob/master/README.md#examples)
+In case configuration is not provided in task manifest, plugin will use default path `/proc/laodavg`
```
Comment (katarzyna-z) [inline README.md]: Typo,  it should be '/proc/loadavg' instead of /'proc/l**ao**davg'

File: README.md (hunk 1 above)
Comment (katarzyna-z) [inline README.md]: After changing 'procfs_path' to 'proc_path' readme wasn't updated.

**Comment by katarzyna-z** [comment]
LGTM

**Pull request merged**